install/source/index.markdown

```diff
@@ -98,6 +98,62 @@ Note, every command to ``catkin config --blacklist`` will override the previous
          moveit_planners_chomp \
          chomp_motion_planner
 
+## Optional: Install a Compiler Cache
+
+Even if you decide to not build *all* of MoveIt, building it from source can still take quite some time.
+Because of this, it can be beneficial to use a tool called a *compiler cache* to avoid recompiling files that have not been changed between builds but which with a regular build (without such a cache) would still be processed by the compiler.
+This situation is especially common when switching between different branches while developing and when `clean`ing a Catkin workspace and then rebuilding it.
+
+[ccache](https://ccache.dev) provides such a compiler cache and installing it can reduce the time needed to rebuild MoveIt (and other packages) from many minutes to mere seconds.
+Note: `ccache` is only compatible with GCC and only caches (Obj)C/(Obj)C++ output, but as MoveIt is largely a C++ application it can benefit greatly.
+
+**Note**: as `ccache` uses the filesystem to cache objects, be sure to have sufficient free space available.
+The default maximum cache size on Debian and Ubuntu is up to 5 GB for a full cache (if there are no files in the cache, it will not take up any space).
+Configure a different maximum with the `--max-size` option.
+
+Refer to the [ccache website](https://ccache.dev) for more information on when the cache is used and how to change its configuration.
+
+### Installation (Debian/Ubuntu)
+
+On a Debian or Ubuntu system, installing `ccache` is simple:
+
+    sudo apt-get install ccache
+
+### Installation (other systems)
+
+On other OS, consult the package manager or software store and search for `ccache`.
+Refer to the [ccache website](https://ccache.dev) for more information on downloading and installation.
+
+### Setup
+
+`ccache` provides a wrapper script that inserts itself between calls to GCC and the cache.
+The script is compatible with GCC command line invocations, but it must be explicitly configured for use (just installing the `ccache` package is not enough).
+
+To use caching with *all* invocations of GCC, it is sufficient to place the `/usr/lib/ccache` directory on the `PATH` (note: `usr/lib/ccache` is used on Debian and Ubuntu systems, adapt paths accordingly on other OS).
+By making sure `ccache` is found *before* the regular GCC, all invocations of GCC will automatically go through the `ccache` wrapper script.
```

Comment:
```suggestion
By making sure ccache's wrapper scripts are found *before* the regular compiler (e.g. GCC), all compiler invocations will automatically go through the `ccache` wrapper script.
```

Reply:
There is actually only a single wrapper, but multiple symlinks to it. So I don't believe I can accept your suggestion, as it's incorrect.

Reply:
Well, you are right. But it's not a script named `ccache` that should be found before the compiler(s), but those symlinks that are named as the compiler(s). :wink:
As you didn't mention the other compiler symlinks before, I'm fine with constraining to GCC here as well.